### ISSUE TITLE
DenseNets in Tensorflow

### ISSUE THREAD

**Issue by Eshan-Agarwal** [tensorflow/models#8278]
Is TensorFlow have `DenseNet `architecture like others `ResNet `etc. Can I add `DenseNet` architecture in `Tensorflow/Models`. I previously worked on it.  Or just load it using `keras `as TF supports keras.

**Comment by saberkun** [comment]
Hi Eshan-Agarwal,
Because DenseNet is inside keras.application and users may just need add some regularization to train a keras.application model, it does not look like an important model to re-implement.
Here is an example: https://github.com/tensorflow/models/blob/master/official/vision/image_classification/resnet_imagenet_main.py#L191
 We do not have plan to add DenseNet at this moment.

**Comment by Eshan-Agarwal** [comment]
@saberkun  Thanks and okay.

**Comment by bhack** [comment]
@Eshan-Agarwal If you are interested ina a Tensorflow2 impl https://github.com/okason97/DenseNet-Tensorflow2 /cc @okason97

**Comment by Eshan-Agarwal** [comment]
@bhack  thank you so much it will help.

**Comment by srilman** [comment]
Not to be rude @saberkun, but the reasoning that "keras.applications has DenseNet so there's no need to reimplement it" seems like a pretty weak argument. ResNet and ResNet_v2 are both implemented in keras.applications but tensorflow/models also has an implementation which IMO is more flexible but not as easy to use.

To be fair, ResNets are much more popular than DenseNets. But at the same time, I do think a more flexible version of DenseNets (as well as less popular models) would be useful for people who have more varied use cases. For example, one could not just use the DenseNet provided in Keras directly to train for CIFAR, even though the original paper does (because the first couple of layers are different).

BTW, would be happy to contribute if its OK!

**Comment by saberkun** [comment]
Hi @srilman, the official models cost more to maintain. We are in progress to build general vision modeling libraries. I think it would be chance to have DenseNet empowered by the library as we also want to bridge the existing object detection and slim model zoo.

Defer to @jaeyounkim for the detailed plan. 
Right now, we establish the community folder. If there is a very good DenseNet TF repository, we could review and recommend it.

**Comment by jaeyounkim** [comment]
@srilman I've added a "community suggestion" label to this issue to keep tracking suggestions from the community.

We will welcome more contributions for new models once we release our new modeling library for computer vision. We will share our roadmap with the community soon.

Please check our [contribution guidelines for paper code](https://github.com/tensorflow/models/wiki/Research-paper-code-contribution) for the time being.

**Comment by bhack** [comment]
@jaewoosong  Are you closing the issues tagged as "community suggestion" or just this one?

**Comment by jaeyounkim** [comment]
I am going to close all the issues labelled as "community suggestion" because I will be tracking "community suggestion" issues separately and revisit them monthly. 

Feel free to reopen it if you have more suggestions.

**Comment by bhack** [comment]
If it is for an internal review plain it is ok but it could cause duplication cause I think that users will not check closed ticket so often.

**Comment by jaeyounkim** [comment]
@bhack I've created a new issue to list all requests for paper implementation. Please create a new issue if you want to request for other paper implementations after checking our [model selection criteria](https://github.com/tensorflow/models/wiki/Research-paper-code-contribution#model-selection).
